### ISSUE TITLE
substitution

### ISSUE THREAD

**Issue by RobinHankin** [RobinHankin/freealg#4]
In the `mvp` package we can do this:
```
 x <- as.mvp("1+3*x*y^2 + y")
 x
mvp object algebraically equal to
1  +  3 x y^2  +  y
 x %>% subs(x="1+e^4")
mvp object algebraically equal to
1  +  3 e^4 y^2  +  y  +  3 y^2
```
The same thing can sort of work in `freealg`.  What I would like is something like this:
```
 x <- as.freealg("1+3yxy+y")
 x %>% subs(25,"2+eeee")  # currently fails but what I want is:
 1 + 3*as.freealg("2+eeee")*as.freealg("x")*as.freealg("2+eeee") + as.freealg("2+eeee")
free algebra element algebraically equal to
 + 3 + 1*eeee + 6*eeeex + 3*eeeexeeee + 12*x + 6*xeeee
``` 

(number 25 in the `subs()` function because `x` is letter number 25)

**Comment by RobinHankin** [comment]
I will work on this on the `subs` branch.

**Comment by RobinHankin** [comment]
Fixed in 16f77a755b5ff808c715ef02d
